minio/api.py

```diff
@@ -824,7 +824,8 @@ def get_partial_object(self, bucket_name, object_name, offset=0, length=0,
                                         sse=sse)
 
     def copy_object(self, bucket_name, object_name, object_source,
-                    conditions=None, source_sse=None, sse=None, metadata=None):
+                    conditions=None, source_sse=None, sse=None, metadata=None,
+                    url_encode=True):
```

Comment:
This is not optional as per AWS S3 @johnnyvf24 - if this to make it work with an S3 compatible server the server should be fixed here.

Reply:
can you also state which is this S3 compatible service? do they GitHub issues?

Reply:
The service is netapp. Would you happen to know of a work-around? Pardon me, I'm not familiar with the Spec, do you happen to have a link that defines it? Thanks

Reply:
There is no workaround @johnnyvf24 https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html

> x-amz-copy-source
> The name of the source bucket and key name of the source object, separated by a slash (/). Must be URL-encoded.

> Pattern: \/.+\/.+

Reply:
The server should be fixed here we cannot carry this baggage in our SDK

Reply:
I see. I'll take it up with NetApp then

Reply:
Actually while looking at that same page via the examples it seems the `/` is not url-encoded. Not that you are wrong, just it seems odd to me.

Reply:
It is not path encoded it is query URL encoded style @johnnyvf24 - that is what they mean by URL encoded style.

Reply:
Those examples are just placeholders what we have implemented is the real server behavior.

Reply:
Didn't notice that, but regardless, it seems this sdk is replacing all `/` with `%2F`? @harshavardhana

Reply:
Correct because python query encoding is broken @johnnyvf24

Reply:
@harshavardhana after looking for examples online I think I have found the issue. It seems the individual components between the `/`s should be uri escaped but the slashes themselves should be untouched. As a basis for this claim, here is an old pull request that does exactly what we are talking about: https://github.com/aws/aws-sdk-js/pull/714/files .

The actual part that used to do the encoding is found here: https://github.com/aws/aws-sdk-js/blob/c2ad3ef8fd486bb58edc225f74152dda51f9f2fe/lib/util.js#L51

Reply:
The line in `compat.py` referenced above replaces all the `/` in the string. I can confirm that when calling NetApp without the `%2F`s that the call works as intended

Reply:
Then please send a PR with the fix we shall take it in.

Reply:
Will do! Not sure how the server side should be handled, so I'll just post one for this SDK.